File: services/multiplication-service/internal/domain/calculation_result.go
package domain

import (
	"time"
	"github.com/shopspring/decimal"
)

type CalculationResult struct {
	Result         decimal.Decimal `json:"result"`
	ProcessingTime time.Duration   `json:"processingTime"`
	Cached         bool            `json:"cached"`
	Timestamp      time.Time       `json:"timestamp"`
}

func NewCalculationResult(result decimal.Decimal, processingTime time.Duration, cached bool) *CalculationResult {
	return &CalculationResult{
		Result:         result,
		ProcessingTime: processingTime,
		Cached:         cached,
		Timestamp:      time.Now(),
	}
}

File: services/division-service/src/domain/calculation_result.rs
use chrono::{DateTime, Utc};
use rust_decimal::Decimal;
use serde::{Deserialize, Serialize};
use std::time::Duration;

#[derive(Debug, Clone, Deserialize, Serialize)]
pub struct CalculationResult {
    pub result: Decimal,
    #[serde(rename = "processingTime")]
    pub processing_time: String, // ISO 8601 duration format
    pub cached: bool,
    pub timestamp: DateTime<Utc>,
}

impl CalculationResult {
    pub fn new(result: Decimal, processing_time: Duration, cached: bool) -> Self {
        Self {
            result,
            processing_time: format!("PT{:.3}S", processing_time.as_secs_f64()),
            cached,
            timestamp: Utc::now(),
        }
    }
}